src/components/Clock.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+interface ClockProps {
+  name: string;
+}
+
+interface ClockState {
+  time: string;
+  hasClock: boolean;
+}
+
+const getDate = () => new Date().toUTCString().slice(-12, -4);
+
+export class Clock extends React.Component<ClockProps, ClockState> {
+  private intervalId?: number;
+
+  state: ClockState = {
+    time: getDate(),
+    hasClock: true,
```

Comment:
you shouldn't have this variable here, it should be only in the App and render entire component based on it